### ISSUE TITLE
Tests fail (with Test::Deep 1.126?)

### ISSUE THREAD

**Issue by eserte** [JaSei/Net-AmazonS3-Simple#3]
My smokers started to report the following failure:
```
Output from './Build test':

Can't call method "isa" on an undefined value at /home/cpansand/.cpan/build/2017010709/Test-Deep-1.126-dGQbw8/blib/lib/Test/Deep.pm line 421.
    # Child (save_object_to_file) exited without calling finalize()

#   Failed test 'save_object_to_file'
#   at /usr/perl5.20.1Dp/lib/site_perl/5.20.1/Test/Builder.pm line 279.
# Looks like you failed 1 test of 3.
# Looks like your test exited with 255 just after 3.
t/Net/AmazonS3/Simple.t ................ 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/3 subtests 
```

Statistical analysis suggests that the problem started with Test::Deep 1.126 (@rjbs: FYI):
```
****************************************************************
Regression 'mod:Test::Deep'
****************************************************************
Name           	       Theta	      StdErr	 T-stat
[0='const']    	      1.0000	      0.0000	5415400008213352.00
[1='eq_0.117'] 	      0.0000	      0.0000	   0.43
[2='eq_1.120'] 	      0.0000	      0.0000	   0.58
[3='eq_1.123'] 	     -0.0000	      0.0000	  -1.17
[4='eq_1.124'] 	      0.0000	      0.0000	   0.00
[5='eq_1.126'] 	     -1.0000	      0.0000	-5360972496103486.00

R^2= 1.000, N= 93, K= 6
****************************************************************
```

**Comment by JaSei** [comment]
```
ok 2 - get_object
    # Subtest: save_object_to_file
Can't call method "isa" on an undefined value at /home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Test-Deep-1.126/blib/lib/Test/Deep.pm line 421.
	Test::Deep::wrap() called at /home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Test-Deep-1.126/blib/lib/Test/Deep.pm line 387
	Test::Deep::descend() called at /home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Test-Deep-1.126/blib/lib/Test/Deep/HashElements.pm line 36
	Test::Deep::HashElements::descend(Test::Deep::HashElements=HASH(0x155e220), HASH(0xc8c5c8)) called at /home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Test-Deep-1.126/blib/lib/Test/Deep.pm line 398
	Test::Deep::descend(HASH(0xc8c5c8), Test::Deep::HashElements=HASH(0x155e220)) called at /home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Test-Deep-1.126/blib/lib/Test/Deep/Hash.pm line 31
	Test::Deep::Hash::descend(Test::Deep::Hash=HASH(0x132ad30), HASH(0xc8c5c8)) called at /home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Test-Deep-1.126/blib/lib/Test/Deep.pm line 398
	Test::Deep::descend(HASH(0xc8c5c8), HASH(0x132aa60)) called at /home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Test-Deep-1.126/blib/lib/Test/Deep.pm line 221
	Test::Deep::cmp_details(HASH(0xc8c5c8), HASH(0x132aa60)) called at /home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Test-Deep-1.126/blib/lib/Test/Deep.pm line 202
	Test::Deep::cmp_deeply(HASH(0xc8c5c8), HASH(0x132aa60), "Net::AmazonS3::Simple::Object::File->create_from_response par"...) called at t/Net/AmazonS3/Simple.t line 29
	main::__ANON__[t/Net/AmazonS3/Simple.t:36]("Net::AmazonS3::Simple::Object::File", "validate", 1, "response", "file_path", Path::Tiny=ARRAY(0xce6f40)) called at /home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Mock-Quick-1.111/blib/lib/Mock/Quick/Class.pm line 114
	Mock::Quick::Class::__ANON__[/home/seidl/.carmel/5.22.3-x86_64-linux-thread-multi/builds/Mock-Quick-1.111/blib/lib/Mock/Quick/Class.pm:114]("Net::AmazonS3::Simple::Object::File", "validate", 1, "response", "file_path", Path::Tiny=ARRAY(0xce6f40)) called at lib/Net/AmazonS3/Simple.pm line 168
	Net::AmazonS3::Simple::save_object_to_file(Net::AmazonS3::Simple=HASH(0xc95e28), "test-bucket", "test-key") called at t/Net/AmazonS3/Simple.t line 63
	main::__ANON__[t/Net/AmazonS3/Simple.t:70]() called at /usr/share/perl5/vendor_perl/Test/Builder.pm line 261
	Test::Builder::__ANON__[/usr/share/perl5/vendor_perl/Test/Builder.pm:264]() called at /usr/share/perl5/vendor_perl/Test/Builder.pm line 266
	eval {...} called at /usr/share/perl5/vendor_perl/Test/Builder.pm line 266
	Test::Builder::subtest(Test::Builder=HASH(0x8c6b88), "save_object_to_file", CODE(0xce3f58)) called at /usr/share/perl5/vendor_perl/Test/More.pm line 771
	Test::More::subtest("save_object_to_file", CODE(0xce3f58)) called at t/Net/AmazonS3/Simple.t line 70
 at /usr/share/perl5/vendor_perl/Test/Builder.pm line 279.
	Test::Builder::subtest(Test::Builder=HASH(0x8c6b88), "save_object_to_file", CODE(0xce3f58)) called at /usr/share/perl5/vendor_perl/Test/More.pm line 771
	Test::More::subtest("save_object_to_file", CODE(0xce3f58)) called at t/Net/AmazonS3/Simple.t line 70
    # Child (save_object_to_file) exited without calling finalize()
```

**Comment by JaSei** [comment]
fixed in version 0.1.3
